share/eds/blockstore.go

```diff
@@ -103,11 +113,11 @@ func (bs *blockstore) HashOnRead(bool) {
 func (bs *blockstore) getReadOnlyBlockstore(ctx context.Context, cid cid.Cid) (dagstore.ReadBlockstore, error) {
 	keys, err := bs.store.dgstr.ShardsContainingMultihash(ctx, cid.Hash())
 	if err != nil {
+		if errors.Is(err, datastore.ErrNotFound) {
+			return nil, ErrNotFound
+		}
 		return nil, fmt.Errorf("failed to find shards containing multihash: %w", err)
 	}
-	if len(keys) == 0 {
-		return nil, errShardNotFound
-	}
```

Comment:
Looks like it suppose to return`datastore.ErrNotFound`. 

https://github.com/ipfs/go-ds-badger/blob/bef422d58aa24884bff3397fe04215ec6ee945dd/datastore.go#L607
```
func (t *txn) get(key ds.Key) ([]byte, error) {
	item, err := t.txn.Get(key.Bytes())
	if err == badger.ErrKeyNotFound {
		err = ds.ErrNotFound
	}
	if err != nil {
		return nil, err
	}

	return item.ValueCopy(nil)
}
```

Reply:
We don't need to check for length because our GetShardsForMultihash will always either return nil or a populated array. This is because there is no functionality yet for removing multihashes from the inverted index, so a key cannot exist that points to an empty array of shard keys. I think the length check was just an artifact from the upstream implementation.

This diff is 👍🏼  but I would prefer to not nest, and just have the `if errors.Is` check above the nil check. But up to vlad, either way is fine for me